File: Phind-CodeLlama-34B-v2/stage2/context_template_full/serial_construct_reduction_clause.f90
#ifndef T1
!T1:serial construct reduction clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER :: i, j, k
        REAL(8) :: a(10), b(10), c(10), d(10)
        REAL(8) :: sum_a, sum_b, sum_c, sum_d

        DO i = 1, 10
          a(i) = i
          b(i) = i * 2
          c(i) = i * 3
          d(i) = i * 4
        END DO

        !$acc serial
        !$acc loop reduction(+:sum_a)
        DO i = 1, 10
          sum_a = sum_a + a(i)
        END DO
        !$acc end serial

        !$acc serial
        !$acc loop reduction(*:sum_b)
        DO i = 1, 10
          sum_b = sum_b * b(i)
        END DO
        !$acc end serial

        !$acc serial
        !$acc loop reduction(max:sum_c)
        DO i = 1, 10
          sum_c = MAX(sum_c, c(i))
        END DO
        !$acc end serial

        !$acc serial
        !$acc loop reduction(min:sum_d)
        DO i = 1, 10
          sum_d = MIN(sum_d, d(i))
        END DO
        !$acc end serial

        IF (sum_a .NE. 55 .OR. sum_b .NE. 38416 .OR. sum_c .NE. 30 .OR. sum_d .NE. 1) THEN
          errors = errors + 1
        END IF

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END If
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: Deepseek-Coder-33b-Instruct/stage2/serial_construct_reduction_clause.cpp
#include "acc_testsuite.h"
#ifndef T1
//T1:serial construct reduction clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    int data[SIZE];
    int sum = 0;

    // Initialize the array with random values
    for(int i = 0; i < SIZE; i++) {
        data[i] = rand() % 100;
    }

    // Compute the sum of the array elements
    #pragma acc serial loop reduction(+:sum)
    for(int i = 0; i < SIZE; i++) {
        sum += data[i];
    }

    if (sum != expected_sum) {
        err++;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}